packages/@aws-cdk/aws-stepfunctions/lib/states/task-base.ts

```diff
@@ -123,14 +139,27 @@ export abstract class TaskStateBase extends State implements INextable {
   protected abstract readonly taskPolicies?: iam.PolicyStatement[];
 
   private readonly timeout?: cdk.Duration;
+  private readonly timeoutSecondsPath?: string;
   private readonly heartbeat?: cdk.Duration;
+  private readonly heartbeatSecondsPath?: string;
   private readonly credentials?: Credentials;
 
   constructor(scope: Construct, id: string, props: TaskStateBaseProps) {
     super(scope, id, props);
+
+    if (props.timeout && props.timeoutSecondsPath) {
```

Comment:
Let's avoid mutually exclusive props here. I'm thinking that we should deprecate the original props and do something here along the lines of the [Schedule](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-events/lib/schedule.ts) class.

Reply:
What should be the name of the new prop?

Reply:
Does `taskTimeout` and `heartbeatTimeout` make sense? Totally open to better ideas.